js/components/graph/Node.js

```diff
@@ -2,57 +2,99 @@ import PropTypes from "prop-types";
 import React from "react";
 import { refLookUp } from "../common/utils";
 
+/** React component class representing a Node on the graph */
 export default class Node extends React.Component {
+  /** Create a node */
   constructor(props) {
     super(props);
     var state = localStorage.getItem(this.props.JSON.id_);
+
+    /**
+     * Unselected states:
+     *  - takeable: all prerequisites are satisfied (or no prereqs/parents)
+     *  - inactive: missing some prerequisites
+     * 
+     * Selected states:
+     *  - active: all prerequisites are satisfied
+     *  - overridden: missing some prerequisites (will have a red border)
+     */
+
     if (this.props.editMode) {
       state = ""; // TODO: define what editMode is
     } else if (state === null) {
+      // sets state to takeable if it has no parent nodes (prereqs)
       state = this.props.parents.length === 0 ? "takeable" : "inactive";
     }
+
     this.state = {
       status: state,
-      selected: ["active", "overridden"].indexOf(state) >= 0
+      selected: ["active", "overridden"].indexOf(state) >= 0 // checks if state means its selected
     };
   }
 
+  /**
+   * Checks whether this Node is selected
+   * @return {boolean}
+   */
   isSelected = () => {
+    // Hybrid nodes: the grey, smaller nodes representing another node farther way
+    // They can only be either active or inactive.
     if (this.props.hybrid) {
       return this.state.status === "active";
     } else {
       return this.state.selected;
     }
   }
 
+  /**
+   * Checks whether all prerequisite/ preceeding nodes for the current one are satisfied
+   * @return {boolean}
+   */
   arePrereqsSatisfied = () => {
-    var svg = this.props.svg;
+    var svg = this.props.svg;  // contains information about the whole graph
+
+    /**
+     * Recursively checks that preceeding nodes are selected
+     * @param  {string|Array} element Node(s)/ other on the graph
+     * @return {boolean}
+     */
     function isAllTrue(element) {
+      // base case: single node
```

Comment:
Remove all inline comments in this function, and the one on line 73; they are just repeating what the code does, and in general we should not include such comments.

Reply:
Should I remove the one next to var svg...?

Reply:
Oh, yes that one too